### DISCUSSION TITLE
How to add nix language support?

### DISCUSSION THREAD

**Discussion by SolidRhino** [LazyVim/LazyVim#2483] [Q&A] [answered]
How can I add nix language support?

- Syntax highlighting
- LSP
- Formatting

And other tools or plugins that are valuable for a nix environment.

**Comment by 0x2E** [comment, marked as answer]
Syntax highlighting is provided by treesitter, nix is in [supported](https://github.com/nvim-treesitter/nvim-treesitter#supported-languages). Try `:TSInstall nix`.

LSP, formatter, etc. can be installed by mason.nvim. Search `language:"nix"` in https://mason-registry.dev/registry/list .
